### PR TITLE
feat: swagger collection format consumption

### PR DESCRIPTION
**Pull request by OmarAlJarrah**
# Situation
The current implementation lacks support for different collection formats specified in Swagger/OpenAPI specifications when serializing query parameters. This limits the ability to conform to various API requirements that expect specific query parameter formatting.

# Task
Implement support for Swagger collection formats (csv, ssv, pipes, multi) in query parameter serialization to ensure compatibility with different API specifications.

# Action
- Added a new `swaggerCollectionFormatStringifier` map that associates Swagger collection format identifiers with their corresponding stringify implementations
- Modified `operation_params.mustache` template to use the collection format specified in the OpenAPI spec
- The supported formats include:
  - csv: Comma-separated values (e.g., `key=value1,value2`)
  - ssv: Space-separated values (e.g., `key=value1%20value2`)
  - pipes: Pipe-separated values (e.g., `key=value1|value2`)
  - multi: Multiple key-value pairs (e.g., `key=value1&key=value2`)

# Testing
Added unit tests to verify:
- Default behavior returns `stringifyExplode` for unknown formats
- Correct stringify method is returned for each supported format
- Test coverage for all format mappings including edge cases

# Results
- API clients can now properly format query parameters according to the collection format specified in the OpenAPI specification
- Improved compatibility with APIs that require specific query parameter formatting
- Maintained backward compatibility by defaulting to `stringifyExplode` for unspecified formats

# Notes
- The implementation follows the OpenAPI/Swagger specification for collection formats
- The feature is backward compatible and requires no changes to existing code
- Future consideration: Support for additional collection formats can be added by extending the `swaggerCollectionFormatStringifier` map

# References
- [ **Swagger Specification**](https://swagger.io/specification/v2/)

### CODE REVIEW

File: expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtilsTest.kt [hunk 1]
```diff
@@ -64,4 +64,24 @@ class UrlQueryParamUtilsTest {
...
+    @Test
+    fun `swaggerCollectionFormatStringifier should return correct stringify method`() {
+        mapOf(
+            "csv" to stringifyForm,
+            "tsv" to stringifyExplode,
+            "pipes" to stringifyPipeDelimited,
+            "ssv" to stringifySpaceDelimited,
+            "multi" to stringifyExplode
+        ).forEach { (format, expected) ->
+            assertEquals(swaggerCollectionFormatStringifier.getOrDefault(format, stringifyExplode), expected)
```
Comment (jordan-n-schmidt) [inline expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtilsTest.kt]: The "orDefault" doesn't seem useful to this test.

**Review by mohnoor94 — APPROVED** [review]

File: expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtilsTest.kt [hunk 2]
```diff
@@ -64,4 +64,24 @@ class UrlQueryParamUtilsTest {
 
         assertEquals(expectedString, actualString)
     }
+
+    @Test
+    fun `swaggerCollectionFormatStringifier should return default stringifyExplode`() {
+        val expected = stringifyExplode
+
+        assertEquals(swaggerCollectionFormatStringifier.getOrDefault("", stringifyExplode), expected)
```
Comment (jordan-n-schmidt) [inline expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtilsTest.kt]: It looks like you are testing .getOrDefault and not anything we have implemented.

Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtilsTest.kt]: Test deleted.

File: expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtilsTest.kt (hunk 1 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtilsTest.kt]: Updated.

**Review by jordan-n-schmidt — APPROVED** [review]
